port_drayage_plugin/src/port_drayage_plugin.cpp

```diff
@@ -66,5 +77,99 @@ namespace port_drayage_plugin
 
         return 0;
     }
+
+    bool PortDrayagePlugin::plan_maneuver_cb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp){
+        
+        geometry_msgs::PoseStamped stop_loc;
+
+        lanelet::BasicPoint2d current_loc(curr_pose_->pose.position.x, curr_pose_->pose.position.y);
+        auto current_lanelets = lanelet::geometry::findNearest(wm_->getMap()->laneletLayer, current_loc, 1);
+
+        if(current_lanelets.size() == 0) {
+            ROS_WARN_STREAM("Cannot find any lanelet in map!");
+            return false;
+        }
+
+        auto current_lanelet = current_lanelets[0];
+        // uncomment after StopRule add.
+
+        // auto stop_rules = current_lanelet.second.regulatoryElementsAs<StopRule>();
+        // if(stop_rules.empty()){
+        //     return NULL;
+        // } else{
+        //     StopRule::Ptr stopRegelem = stop_rules.front();
+        //     // TODO get stopRegelem position and return the object with downtrack
+        //     stop_loc = geometry_msgs::PoseStampedConstPtr pose;
+        // }
+
+        // TODO
+        double stop_loc_downtrack = 0;
+        double current_loc_downtrack = wm_->routeTrackPos(current_loc).downtrack;
```

Comment:
You should probably use the downtrack `end_dist` value from the last maneuver in the maneuver list in the `req` or the vehicle's current position if there are no maneuvers in the request list. We're basically trying to plan what we'd like to do after the maneuvers in the request list have executed, so imagine we're at the end location of the last maneuver at the specified time, what would we do from there?